crates/web-sys/tests/wasm/main.rs

```diff
@@ -56,3 +57,11 @@ pub mod table_element;
 pub mod title_element;
 pub mod xpath_result;
 pub mod indexeddb;
+
+#[wasm_bindgen_test]
+fn deref_works() {
+    let x = JsValue::from(3);
```

Comment:
Maybe do 

```rust
let x = js_sys::Object::new();
let y: &JsValue = &x;
```

so that we aren't doing these unchecked casts that would fail if they were checked?